game_data.gd

```diff
@@ -1,41 +1,60 @@
 class_name GameData
 extends Node
 
-enum TETROMINO { square, tee, l_left, l_right, zag_right, zag_left, line , double_square }
+enum Tetromino {
+    SQUARE,
+    TEE,
+    L_LEFT,
+    L_RIGHT,
+    ZAG_RIGHT,
+    ZAG_LEFT,
+    LINE,
+    DOUBLE_SQUARE,
+}
 
-const cosine = cos(PI / 2)  ## = 0
-const sine = sin(PI / 2)  ## = 1
-const rotation_matrix = [cosine, sine, -sine, cosine]
+const COSINE = cos(PI / 2)  ## = 0
+const SINE = sin(PI / 2)  ## = 1
+const ROTATION_MATRIX = [COSINE, SINE, -SINE, COSINE]
 
-const tile_ids = {
-    TETROMINO.square: 0,
-    TETROMINO.tee: 1,
-    TETROMINO.l_left: 2,
-    TETROMINO.l_right: 3,
-    TETROMINO.zag_right: 4,
-    TETROMINO.zag_left: 5,
-    TETROMINO.line: 6, 
-    TETROMINO.double_square: 0
+const TILE_ID = {
+    Tetromino.SQUARE: 0,
+    Tetromino.TEE: 1,
+    Tetromino.L_LEFT: 2,
+    Tetromino.L_RIGHT: 3,
+    Tetromino.ZAG_RIGHT: 4,
+    Tetromino.ZAG_LEFT: 5,
+    Tetromino.LINE: 6,
+    Tetromino.DOUBLE_SQUARE: 0,
```

Comment:
@rp282 Why does DOUBLE_SQUARE have the same ID as SQUARE?

Reply:
These are the tile ids used to pick the colors of the tetrominos. We'll need to decide if our added pieces have their own tiles/

Reply:
I think we should have a separate list for piece indices and tile IDs (colors):
- Piece index: a unique identifier for each piece
- Tile ID: maps to the tile ID map

This way, it'll be possible for different pieces to have the same tile IDs i.e. square & double square with tile ID 0. Else we run into situations where we can't disambiguate pieces i.e. piece holding.